gr-blocks/lib/tcp_connection.cc

```diff
@@ -54,16 +54,25 @@ namespace gr {
     tcp_connection::send(pmt::pmt_t vector)
     {
       size_t len = pmt::blob_length(vector);
+
+      // Asio async_write() requires the buffer to remain valid until the handler is called. It will be
+      // deleted in tcp_connection::handle_write.
+      char* txbuf = new char[len];
```

Comment:
I'm fairly sure this is okay, but not knowing asio internals, it's not clear if there are times that async_write() might be called but without a following handle_write being invoked.